### ISSUE TITLE
Many 'string "Module:Quotations"' error messages

### ISSUE THREAD

**Issue by ChristianSi** [tatuylonen/wiktextract#164]
Since we few weeks, the translations lists in the English dump include various error messages that start like '[string "Module:Quotations"]:118: Please specify a language code in the first parameter. The value...'.

Examples include:

* Italian translations of "Americas"
* German translations of "bald"
* Portuguese translations of "cheap"

And several thousand others.

It would be great if translation parsing could be made more robust to avoid this error.

**Comment by kristian-clausal** [comment]
For some reason, it seems {{q}}-templates are being expanded as {{Q}} templates, which leads to the parameter error. I will try to take a look if we're messing with template name capitalization somewhere, but this is something I'm not familiar with yet so we'll see; hopefully Oskari or Tatu might know something.

**Comment by xxyzz** [comment]
This error should be fixed by https://github.com/tatuylonen/wikitextprocessor/pull/13/commits/a011dc4f7b126c473bb116d2c0144466b8956893 once the pull request is merged.

**Comment by xxyzz** [comment]
It's inside the https://github.com/tatuylonen/wikitextprocessor/pull/13 pull request, GitHub changed the commit link...

**Comment by kristian-clausal** [comment]
While we're waiting for that, in the meanwhile I disabled
`        self.templates[name.lower()] = body
`
in core.py, which was added in the last merge related to the above Chinese wiktionary stuff, and it did not change the errors. I also did not get the erroneous data this morning, and then I noticed I had an old dump; swapping to the new dump from the 1st caused the errors to appear, so @xxyzz please make sure to use the new dump when testing your fixes.

That the errors appeared after the data was updated is annoying, because it's still obviously a bug on our side; unless there were a ton of script errors for a hot minute when the dump was created and a hotfix on wiktionary's side was too slow to get included in the newest dump, then there's still a discrepancy between our output and wiktionary's output.

**Comment by kristian-clausal** [comment]
@xxyzz I branched off the old commit 4a539600dd5ed1828c064f47c5d5673ba2cd6af9 before the merge, and it doesn't fix the issue in this thread; it's not a problem with your code. Hurray! Oh no.

**Comment by xxyzz** [comment]
https://github.com/tatuylonen/wikitextprocessor/commit/4a539600dd5ed1828c064f47c5d5673ba2cd6af9 is after https://github.com/tatuylonen/wikitextprocessor/commit/0f845269f75307dad4969b6fdaa8991b4c91bce4,  https://github.com/tatuylonen/wikitextprocessor/blob/f172086b19529a1762a7a77130f5265989586a84/wikitextprocessor/core.py#L598 is still there.

**Comment by kristian-clausal** [comment]
@xxyzz The timeline is a bit weird to read because of the merge. When I branched off, I branched before your code was merged into main, even though you had committed it locally before (and even though they 'show' up in the timeline before). The merge "hadn't happened" yet; I'm looking at the code I branched off, and your commits are not there.

**Comment by xxyzz** [comment]
Oh, OK. Have you also clear the cache files? The command hasn't finished running on my machine so I not totally sure that commit fixes the error.

**Comment by kristian-clausal** [comment]
I completely forgot about the cache. I'll regenerate it tomorrow, and see what happens.

**Comment by kristian-clausal** [comment]
After regenerating the cache using the branch with pre-merged code, the output is correct, so I was wrong. While we're waiting for a review on the bigger pull request from @xxyzz, could you patch the relevant code (lang-code parameters at least, but I have no idea what else that might lurk in the background) in the main branch and make a pull request as just a hotfix?

**Comment by xxyzz** [comment]
IMO, remove that line or checkout the working commit is easier for a hotfix. But I think running testing code on production site is a more serious problem...

**Comment by tatuylonen** [comment]
I'm unable to reproduce this now (20221001 dump) with any of the words Americas, bald, cheap.  Is this now fixed?

**Comment by ChristianSi** [comment]
It's not fixed, rather the situation seems unchanged. In the latest kaikki.org-dictionary-English.json.gz dump (from 5 Oct, it seems) I still count about 3000 'string "Module:Quotations"' errors. You can also see it on the website, e.g. in the entries I mentioned:

* [Americas](https://kaikki.org/dictionary/All%20languages%20combined/meaning/A/Am/Americas.html) – Corsican and Italian translations
* [bald](https://kaikki.org/dictionary/All%20languages%20combined/meaning/b/ba/bald.html) – German translation of 1. and 3.
* [cheap](https://kaikki.org/dictionary/All%20languages%20combined/meaning/c/ch/cheap.html) – multiple translations of the first adjective

**Comment by kristian-clausal** [comment]
The website update finally went through an hour after you posted, please check it out if there are still issues lurking. On Oct 8, 2022 17:00, Christian Siefkes ***@***.***> wrote:

It's not fixed, rather the situation seems unchanged. In the latest kaikki.org-dictionary-English.json.gz dump (from 5 Oct, it seems) I still count about 3000 'string "Module:Quotations"' errors. You can also see it on the website, e.g. in the entries I mentioned:

Americas – Corsican and Italian translationsbald – German translation of 1. and 3.cheap – multiple translations of the first adjective

**Comment by kristian-clausal** [comment]
The issue at hand seems to now be fixed. The German translation 3. for bald has a dangling ", etc.)", but that's a different issue, so I'm closing this now. As always, if something pops up, please feel free to open a new thread.

**Comment by ChristianSi** [comment]
Yeah, it's fixed now, thanks a lot!
